From fdde0a6086cfc23be57c6423e6a76238574cfd1a Mon Sep 17 00:00:00 2001
From: Dinolek <git@dinolek.me>
Date: Sat, 12 Jul 2025 18:09:49 +0200
Subject: [PATCH 098/115] arm64: dts: mediatek: mt6785-xiaomi-begonia: Add
 volume keys

Add gpio keys node with volume up and down keys.

Signed-off-by: Dinolek <git@dinolek.me>
---
 .../dts/mediatek/mt6785-xiaomi-begonia.dts    | 34 +++++++++++++++++++
 1 file changed, 34 insertions(+)

diff --git a/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia.dts b/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia.dts
index 81045676c..d12617f10 100644
--- a/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia.dts
+++ b/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia.dts
@@ -4,6 +4,8 @@
  */
 
 /dts-v1/;
+#include <dt-bindings/input/input.h>
+
 #include "mt6785.dtsi"
 #include "mt6359.dtsi"
 
@@ -18,6 +20,28 @@ vibrator {
 		max-microvolt = <2800000>;
 		min-microvolt = <1200000>;
 	};
+
+	volume_buttons: volume-buttons {
+		compatible = "gpio-keys";
+		pinctrl-names = "default";
+		pinctrl-0 = <&volume_button_pins>;
+
+		button-volume-down {
+			label = "Volume Down";
+			linux,code = <KEY_VOLUMEDOWN>;
+			debounce-interval = <100>;
+
+			gpios = <&pio 60 GPIO_ACTIVE_LOW>;
+		};
+
+		button-volume-up {
+			label = "Volume Up";
+			linux,code = <KEY_VOLUMEUP>;
+			debounce-interval = <100>;
+
+			gpios = <&pio 59 GPIO_ACTIVE_LOW>;
+		};
+	};
 };
 
 &cci {
@@ -101,6 +125,16 @@ &mt6359_vsram_others_ldo_reg {
 	regulator-always-on;
 };
 
+&pio {
+	volume_button_pins: volume-button-pins {
+		pins-volume {
+			pinmux = <PINMUX_GPIO59__FUNC_KPCOL0>,
+				 <PINMUX_GPIO60__FUNC_KPCOL1>;
+			input-enable;
+		};
+	};
+};
+
 &pmic {
 	interrupts-extended = <&pio 193 IRQ_TYPE_LEVEL_HIGH>;
 };
-- 
2.56.0

